Assets/BossRoom/Scripts/Client/UI/UIDebugCheats.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using UnityEngine;
+
+#if UNITY_EDITOR || DEVELOPMENT_BUILD
+namespace Unity.Multiplayer.Samples.BossRoom.Client
+{
+    /// <summary>
+    /// Provides a way to open/close a popup window and GameEvents to raise debug cheat commands to be caught by
+    /// listeners. This class is only available in the editor or for development builds
+    /// </summary>
+    public class UIDebugCheats : MonoBehaviour
+    {
+        [SerializeField]
+        GameObject m_DebugCheatsPanel;
```

Comment:
My understanding is you're putting all those events in between because of the client/server separation correct?
What do you think of putting everything in Shared and hooking up the buttons directly to the RPCs? It'd make things way simpler and we could put this cheat code in a self contained "cheat assembly", something like that

Reply:
> My understanding is you're putting all those events in between because of the client/server separation correct?
Yes, that and to keep the UI and logic separate.

> What do you think of putting everything in Shared and hooking up the buttons directly to the RPCs?
It would work for the cheats that are in this PR, but we would lose access to everything in the Server assembly. This could be a problem for some new cheats as well. For example, my current implementation of God Mode uses PlayerServerCharacter to find the character on which to apply the toggle. I could move it to a new assembly with a dependency on Server and Shared, though. It would remove the need for events